tests/test_numpy_array.cpp

```diff
@@ -275,10 +275,16 @@ TEST_SUBMODULE(numpy_array, sm) {
     // [workaround(intel)] ICC 20/21 breaks with py::arg().stuff, using py::arg{}.stuff works.
 
     // Only accept the exact types:
-    // NOLINTNEXTLINE(performance-unnecessary-value-param)
-    sm.def("overloaded3", [](py::array_t<int>) { return "int"; }, py::arg{}.noconvert());
-    // NOLINTNEXTLINE(performance-unnecessary-value-param)
-    sm.def("overloaded3", [](py::array_t<double>) { return "double"; }, py::arg{}.noconvert());
+    sm.def(
+        "overloaded3",
+        // NOLINTNEXTLINE(performance-unnecessary-value-param)
```

Comment:
These probably could const refs to make it happy.

Reply:
Done here and in tests/test_numpy_array.cpp, but systematically in both files.

But I want to propose to do this even more systematically: I volunteer to do this everywhere, in one PR.

Reason: picking out a place here and there, just because the associated original source code line happens to be longer than 99 characters, is really random. If people look later they will wonder why some are `const T &` while others are `T` with the `NOLINT` comment, and possibly read something into it. That's really misleading.

It was nice to use the `NOLINT` to enable activating the clang-tidy option without having to get into the cleanup, but I think doing the cleanup sooner rather than later is best.

If you agree I'll do the systematic `NOLINT` reduction*** first, then come back to this PR after merging and rebasing.

*** I'm assuming there could be a couple exceptions.

Reply:
Okay sounds good. It's not in too many places I suppose.